pkg/cloud/services/secretsmanager/secret_fetch_script.go

```diff
@@ -231,6 +231,7 @@ if [ ${GUNZIP_RETURN} -ne 0 ]; then
 fi
 
 log::info "restarting cloud-init"
-systemctl restart cloud-init
+rm -rf /var/lib/cloud/instances
+cloud-init clean --reboot
```

Comment:
```suggestion
reboot
```

Reply:
We tried the following that worked without reboot:

```
rm -rf /var/lib/cloud/instances
cloud-init clean
systemctl restart cloud-init-local
systemctl restart cloud-init
systemctl restart cloud-config
systemctl restart cloud-final
```